src/components/AnonymousChat/child/ChatItemAttachment.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-unexpected-multiline */
+import * as React from 'react';
+import {Linking, Text, TouchableOpacity, View} from 'react-native';
+import FastImage from 'react-native-fast-image';
+import {useNavigation} from '@react-navigation/core';
+
+import IconFile from '../../../assets/icon/IconFile';
+import IconVideoPlay from '../../../assets/icon/IconVideoPlay';
+import {formatBytes} from '../../../utils/string/StringUtils';
+import {ChatItemAttachmentStyles} from './ChatItemAttachmentStyles';
+
+const ChatItemAttachment = ({attachments = []}: any) => {
+  const navigation = useNavigation();
+
+  const onOpenMediaPreview = (medias, index) => {
+    if (medias.find((media) => media.type === 'video')) {
+      navigation.push('VideoViewer', {
+        title: 'Video',
+        url: medias[index].video_path
+      });
+    } else {
+      navigation.push('ImageViewer', {
+        title: 'Photo',
+        index,
+        images: medias
+          .filter((media) => media.type === 'image')
+          .map((media) => ({url: media.asset_url}))
+      });
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onOpenMediaPreview` function is directly using the `navigation` object to navigate to different screens based on the media type. This tightly couples this component with the navigation logic and makes it less reusable. Consider passing a callback function as a prop to handle media preview requests, which would improve the separation of concerns.

```diff
-  const onOpenMediaPreview = (medias, index) => {
-    if (medias.find((media) => media.type === 'video')) {
-      navigation.push('VideoViewer', {
-        title: 'Video',
-        url: medias[index].video_path
-      });
-    } else {
-      navigation.push('ImageViewer', {
-        title: 'Photo',
-        index,
-        images: medias
-          .filter((media) => media.type === 'image')
-          .map((media) => ({url: media.asset_url}))
-      });
-    }
-  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->